lua/gitportal/git.lua

```diff
@@ -49,8 +49,7 @@ function M.can_open_current_file()
     return false
   end
 
-  local status = cli.run_command("git status")
-  if status == nil then
+  if not vim.fs.root(0, "git") then
```

Comment:
there is no reason to call `git` here, we can check for `.git` directory which is infinitely faster as we don't spawn git process

Reply:
This doesn't seem to be working for me locally. Have you tested this? Should vim.fs.root be searching for `.git`?

Reply:
@tronikelis 
> That command searches from the current buffer 0 and up, so maybe you are editing a buffer that is out of git cwd, if you want to keep old behavior you can do vim.fs.root(vim.uv.getcwd(), ".git") this will start the search from the cwd afaik

Hmm, this doesn't seem as resilient to me. I'm getting some issues locally testing with it. I think actually we could just use `get_git_root_dir` which is defined above. That seems to be working for me and is very resilient so a user can still use it even if they opened nvim in a subfolder below the base git directory. It also returns nil if none is found, so we shouldn't have to make any changes to it.

Reply:
> Hmm, this doesn't seem as resilient to me.

Maybe I wasn't clear enough but, by "out of git cwd" I meant not *under* git cwd

from the help page

```
vim.fs.root({source}, {marker})                                *vim.fs.root()*
    Find the first parent directory containing a specific "marker", relative
    to a file path or buffer.

```

Reply:
This accomplishes the same task as `M.get_git_root_dir()`, but in a more efficient and readable manner. Github wouldn't let me suggest a commit to `M.get_git_root_dir()` because it wasn't close enough to your changes but what I did was 

- Update `M.get_git_root_dir()` to use `vim.fs.root`
- Replace the current call to `vim.fs.root` with a call to `M.get_git_root_dir`, that way we can reuse the existing function and the existing function is improved.

```suggestion
  if not M.get_git_root_dir() then
```